### ISSUE TITLE
Move Schemes and Templates into Their Own Repositories

### ISSUE THREAD

**Issue by alexbooker** [base16-builder/base16-builder#126]
As you already know, this project is based on [**@chriskempson**'s original Base16 Builder](https://github.com/chriskempson/base16-builder). My main aim when starting this project was to rejuvenate @chriskempson's tool, which hadn't been updated in months. 

Recently, @chriskempson got in touch with @aloisdg and I to speak about the future of our projects.

Despite @chriskempson  wanting  to take the project in a different direction to @aloisdg and I, we have ultimately decided to work together.

We're still trying to figure out the precise direction to take, but what we _have_ decided is that **we should move schemes and templates outside the main repository.**

The fact of the matter is that @chriskempson, @aloisdg and I spend most of our time managing pull requests and issues relating to templates we do not use. Because we have little or no experience with most templates, it's hard to ensure that they are up-to-date and of a high quality. Moreover, taking responsibility for so many templates and schemes is time consuming and generally, difficult to manage. We believe that moving all schemes and templates into their own repositories, and allowing their authors to take responsibility for them, will increase their quality; make it easier to consume the templates from other tools; and, allow us to focus on core issues in this repository.

We are convinced this change needs to happen but we are yet to decide on the best approach. 

From what I can gather, @chriskempson's idea is to create a central index for [scheme](https://github.com/chriskempson/base16-schemes-source) and [template](https://github.com/chriskempson/base16-templates-source) repositories. I think this is a great starting point but I would love to hear your thoughts.

**Comment by cruor99** [comment]
I have a suggestion for the template and scheme stuff.
In python, there's a tool called [cookiecutter](https://github.com/audreyr/cookiecutter) - which I use for boostrapping the setup of my Flask projects. 

With that tool, you can either use a local folder - or a remote github project that is [set up the proper way](https://github.com/JackStouffer/cookiecutter-Flask-Foundation). This completely seperates any builder from the template, and also doesn't need a central index or repository for schemes and templates - although it could still be a nice thing.

**Comment by belak** [comment]
Ah, I must have missed this when I opened #131.

What template languages would be an option? We'd need to pick a template language which works across languages and/or pick a subset of templates to use.

It seems like mustache would be the best option if we want to support the most languages, but that also forces us to remove all logic from templates (which I know was a pain in some instances).

As far as I can tell, there aren't very many cross-language template languages, so we're either limited by the logic, or limiting which languages to support.

**Comment by chriskempson** [comment]
I've an example implementation at https://github.com/chriskempson/base16-builder-php. I'm making use of Mustache at the moment.

**Comment by alexbooker** [comment]
@belak I agree that to use a language-agnostic templating engine _like_ Mustache would be ideal.

I'm concerned about templates that require language-specific expressions, however. Creating a specific template for each language is a bit unwieldly in my opinion. The alternative, which is to hard-code [`context`](https://github.com/base16-builder/base16-builder/blob/master/lib/index.js#L6) properties, will also become unwieldy in time, so I'm not sure what to do :confused:.

**Comment by chriskempson** [comment]
I'm also concerned about this. Do we have a list of all the templates that have language-specific expressions? Just wondering how many there are.

**Comment by belak** [comment]
There was some work in the base16-builder/base16-builder repo to catalog some of this... some of them were not ported though.

I believe it was in #23

**Comment by chriskempson** [comment]
Thanks! I see four references to ruby code in that list so I guess there are not too many :)

**Comment by aloisdg** [comment]
@chriskempson Some templates like smartftp had more than one issue. I wrote the first encountered. We will have to check all the end of the list :)

**Comment by chriskempson** [comment]
@aloisdg Thanks for the link.

**Comment by CrashyBang** [comment]
So I was thinking about this the other day and the [vue-cli](https://github.com/vuejs/vue-cli) handles custom/external templates really well: `vue init egoist/vuepack project` where [egoist/vuepack](https://github.com/egoist/vuepack) is the repo for the template.

I was thinking something like this would work really well for base16-builder, I am envisioning something like this: `base16-builder -s crashybang/monokai -t crashybang/vim -b dark > theme`.

It is also worth noting the that the [vue-cli](https://github.com/vuejs/vue-cli) caches the template after the on the first use and as far as I am aware checks for updates next time you use it if an internet connection is available.

The only downside is you need an internet connection every time you want a new template theme but that is still going to be an issue with moving them all out to isolated repo's anyway.

Thoughts?
Cheers.
